front/src/modules/timesStops/helpers/__tests__/utils.spec.ts

```diff
@@ -329,6 +329,144 @@ describe('updateDaySinceDeparture', () => {
       ];
       expect(result).toEqual(expected);
     });
+    it('should handle exactly midnight', () => {
+      const pathWaypointRows = [
+        {
+          opId: 'd9c92cb4',
+          name: 'Ge',
+          uic: 86,
+          ch: 'BV',
+          arrival: { time: '23:50:00' },
+        },
+        {
+          opId: 'd9b38600',
+          name: 'Ge',
+          uic: 86,
+          ch: 'BX',
+          arrival: { time: '00:00:00' },
+        },
+      ] as TimesStopsInputRow[];
+      const startTime = '2024-08-13T23:50:00';
+      const result = updateDaySinceDeparture(pathWaypointRows, startTime, {
+        keepFirstIndexArrival: true,
+      });
+      const expected = [
+        {
+          opId: 'd9c92cb4',
+          name: 'Ge',
+          uic: 86,
+          ch: 'BV',
+          arrival: { time: '23:50:00', daySinceDeparture: 0 },
+          departure: undefined,
+        },
+        {
+          opId: 'd9b38600',
+          name: 'Ge',
+          uic: 86,
+          ch: 'BX',
+          arrival: { time: '00:00:00', daySinceDeparture: 1, dayDisplayed: true },
+          departure: undefined,
+        },
+      ];
+      expect(result).toEqual(expected);
+    });
+    it('should handle departure exactly at midnight', () => {
+      const pathWaypointRows = [
+        {
+          opId: 'd9a382bc',
+          name: 'St',
+          uic: 75,
+          ch: 'BV',
+          arrival: { time: '00:00:00' },
+        },
+        {
+          opId: 'd9b38600',
+          name: 'Ge',
+          uic: 86,
+          ch: 'BX',
+          arrival: { time: '01:00:00' },
+        },
+      ] as TimesStopsInputRow[];
+      const startTime = '2024-08-13T00:00:00';
+      const result = updateDaySinceDeparture(pathWaypointRows, startTime, {
+        keepFirstIndexArrival: true,
+      });
+      const expected = [
+        {
+          opId: 'd9a382bc',
+          name: 'St',
+          uic: 75,
+          ch: 'BV',
+          arrival: { time: '00:00:00', daySinceDeparture: 1, dayDisplayed: true },
+          departure: undefined,
+        },
+        {
+          opId: 'd9b38600',
+          name: 'Ge',
+          uic: 86,
+          ch: 'BX',
+          arrival: { time: '01:00:00', daySinceDeparture: 1 },
+          departure: undefined,
+        },
+      ];
```

Comment:
Missed that earlier, but this test is not correct. both steps should be in day 0 and not display D+

```suggestion
      const expected = [
        {
          opId: 'd9a382bc',
          name: 'St',
          uic: 75,
          ch: 'BV',
          arrival: { time: '00:00:00', daySinceDeparture: 0 },
          departure: undefined,
        },
        {
          opId: 'd9b38600',
          name: 'Ge',
          uic: 86,
          ch: 'BX',
          arrival: { time: '01:00:00', daySinceDeparture: 0 },
          departure: undefined,
        },
      ];
```

![Screenshot_20241002_140303](https://github.com/user-attachments/assets/c222fcb9-c76e-4c72-be76-51a6048cf8b3)

Reply:
You can move the test in the 1-day span section

Reply:
Done + I've modified the function to take this case into account